modules/engine/src/listeners.ts

```diff
@@ -650,10 +670,42 @@ async function handleConditionalTransferCreation(
   };
   evts[EngineEvents.CONDITIONAL_TRANSFER_CREATED].post(payload);
 
-  // If we should not route the transfer, do nothing
-  if (!routingId || transfer.meta?.routingId !== routingId) {
-    logger.warn({ transferId, routingId, meta: transfer.meta }, "Cannot route transfer");
-    return;
+  // TRANSFER_ROUTING_COMPLETE event
+
+  // if transfer is created and we are the end recipient, emit event now and end
```

Comment:
Another double check - 'emit event now and end' - should we be returning in this block (based on that phrasing)?

Reply:
That's correct, fixed.